✨ style: Update Messages Style to Match ChatGPT (ChatGPT UI)

techwithanirudh opened the pull request:
## Summary

Modernized the message components to match ChatGPT’s style and optimized dark mode support.
> Note: ⚠️ This P.R. is one of multiple smaller P.R.s split from a larger one by GPT-4o. 

### Desktop
Before:
![image](https://github.com/danny-avila/LibreChat/assets/70373803/42b39b96-8a6a-407f-96f8-f1eafc7928ba)

After:
![image](https://github.com/danny-avila/LibreChat/assets/70373803/ac58b46c-f588-416d-84da-a24c6ebc905e)

Before:
![image](https://github.com/danny-avila/LibreChat/assets/70373803/0604fc98-3f04-4085-83ea-933d9df98ada)

After:
![image](https://github.com/danny-avila/LibreChat/assets/70373803/42a1aaf3-bd3f-4513-b338-cf249519928f)


### Mobile
Before:
<img width="377" alt="image" src="https://github.com/danny-avila/LibreChat/assets/70373803/bc62e9c5-9740-483d-8baa-df4790d7f30d">

After:
<img width="377" alt="image" src="https://github.com/danny-avila/LibreChat/assets/70373803/124aa4f2-1f74-4344-b0e6-8d28d92b567e">

Before:
<img width="377" alt="image" src="https://github.com/danny-avila/LibreChat/assets/70373803/52a858fb-a7b9-4cfd-bae9-a8bfc2fdd31c">

After:
<img width="377" alt="image" src="https://github.com/danny-avila/LibreChat/assets/70373803/b9f6abb4-c122-44fb-8a1a-8513f28506a9">


## Change Type

- [x] New feature (non-breaking change which adds functionality)

## Testing

N/A

### **Test Configuration**:

N/A

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] My changes do not introduce new warnings
- [x] Local unit tests pass with my changes

techwithanirudh committed:
Updating Messages feature with files from new/feat/new-ui

techwithanirudh committed:
Merge branch 'main' into new/feat/messages

techwithanirudh committed:
Merge branch 'main' into new/feat/messages

techwithanirudh committed:
Merge branch 'main' into new/feat/messages

techwithanirudh commented:
@danny-avila Could you please review this P.R?

berry-13 commented:
@techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?

techwithanirudh committed:
Merge branch 'main' into new/feat/messages

techwithanirudh committed:
feat: MessagesUISwitch (p1)

Note: This code has not been run and is assumed to work, p2 will check if this implementation works and use the recoil state to make modifcations to the UI

techwithanirudh committed:
fix: use key messagesUI instead of autoScroll
fix: use default value as true for messagesUI
fix: add com_nav_messages_ui for the text for the switch

techwithanirudh committed:
feat: quick and dirt implementation

Note: Not tested

techwithanirudh committed:
fix: eslint linting

techwithanirudh committed:
fix: hide buttons by default and show them when on hover (like the default behaviour in previous versions)

techwithanirudh committed:
fix: on mobile devices the HoverButtons are displayed without hover

techwithanirudh committed:
fix: tests from failing (messageUI -> messagesUI)

techwithanirudh commented:
> @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?

Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.

techwithanirudh committed:
fix: tests

Note: As messagesUI is already true expect false

techwithanirudh committed:
feat: Update edit message to take into account messagesUI

techwithanirudh committed:
feat: mimc the EditMessage style in light mode

techwithanirudh committed:
fix: nav-links spacing

berry-13 commented:
> > @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?
> 
> Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.

thanks!

techwithanirudh committed:
Merge branch 'main' into new/feat/messages

danny-avila commented:
> > @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?
> 
> Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.



> > @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?
> 
> Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.



> > @techwithanirudh many of my friends don't want to use the new "Chat bubble". Could we create a switch for it? additionally, can we include this "var" in the "edit messages" PR?
> 
> Once this pull request is merged, I will add the conditional check for the parameter in the EditMessages pull request, but other than that everything is done.

I won't be merging this PR unless there is a way to select different themes for the new UI updates, with the current theme being the default.

techwithanirudh commented:
@danny-avila This has already been implemented in this P.R

danny-avila commented:
> @danny-avila This has already been implemented in this P.R, if this P.R gets merged I could update the others to do the same.

Ok, I didn't realize that. I will try it out.

techwithanirudh commented:
> > @danny-avila This has already been implemented in this P.R, if this P.R gets merged I could update the others to do the same.
> 
> Ok, I didn't realize that. I will try it out.

@danny-avila Thanks! I'll resolve the merge conflicts

techwithanirudh committed:
Merge branch 'main' of https://github.com/danny-avila/Librechat into new/feat/messages

techwithanirudh committed:
feat: use new ui in wordigns

techwithanirudh committed:
style: use black and white colors for link button

danny-avila commented on client/src/store/settings.ts:
```diff
@@ -50,6 +50,7 @@ const staticAtoms = {
 // Atoms with localStorage
 const localStorageAtoms = {
   autoScroll: atomWithLocalStorage('autoScroll', false),
+  messagesUI: atomWithLocalStorage('messagesUI', true),
```
default value should be `false`

danny-avila commented on client/src/localization/languages/Eng.ts:
```diff
@@ -479,6 +479,7 @@ export default {
   com_nav_welcome_assistant: 'Please Select an Assistant',
   com_nav_welcome_message: 'How can I help you today?',
   com_nav_auto_scroll: 'Auto-Scroll to latest message on chat open',
+  com_nav_messages_ui: 'Enable new UI',
```
"Use Bubble styling for User Messages"

danny-avila commented on client/src/components/Nav/SettingsTabs/General/General.tsx:
```diff
@@ -163,6 +164,9 @@ function General() {
         <div className="border-b pb-3 last-of-type:border-b-0 dark:border-gray-600">
           <AutoScrollSwitch />
         </div>
+        <div className="border-b pb-3 last-of-type:border-b-0 dark:border-gray-600">
```
This should go in a new Tab called "Appearance"

danny-avila commented:
The hover buttons should also hover when the entire message row is being hovered in the new style. currently it does not, and it doesn't feel intuitive to hover over the bubble first.


https://github.com/danny-avila/LibreChat/assets/110412045/52b0ac5f-9f86-45e3-a215-9ca2f2e66728

techwithanirudh committed:
feat: Add appearance tab

techwithanirudh committed:
fix: use default value false

techwithanirudh committed:
fix: Add appearance icon

techwithanirudh commented:
> The hover buttons should also hover when the entire message row is being hovered in the new style. currently it does not, and it doesn't feel intuitive to hover over the bubble first.
> 
>  Recording.2024-06-08.183608.mp4

@danny-avila I tried to fix this issue, is this fix okay?

techwithanirudh committed:
Merge branch 'main' into new/feat/messages

techwithanirudh committed:
feat: Use ishovered for messages

techwithanirudh committed:
fix: use py-[16px] for spacing

techwithanirudh committed:
fix: tests to call with true

danny-avila commented:
This PR was closed due to fundamental changes made to messages to accommodate this PR https://github.com/danny-avila/LibreChat/pull/3191

I'm still open to the bubble message look as an option but with geared for dependency injection from the start, with minimal code edits to the main message components and Hover buttons.

I closed this PR as I don't expect the changes to be reconciled from the current state of the PR which already had some merge conflicts before #3191
